actions/download_action.go

```diff
@@ -78,8 +78,23 @@ func (d *DownloadAction) Run(context *debos.DebosContext) error {
 	}
 
 	if d.Unpack == true {
+		archive, err := debos.NewArchive(filename)
+		if err != nil {
+			return err
+		}
+
+		if len(d.Compression) > 0 {
+			if err := archive.SetCompression(d.Compression); err != nil {
+				return err
+			}
+		}
 		targetdir := filename + ".d"
-		err := debos.UnpackTarArchive(filename, targetdir, d.Compression, "--no-same-owner", "--no-same-permissions")
+		switch archive.GetType() {
+		case debos.Tar:
+			err = archive.Unpack(targetdir, "--no-same-owner", "--no-same-permissions")
```

Comment:
Hrm, this requires the user to know that underlying to it all is unix tar. It would be better to have give the Tar archive class a method to turn of the  usage of same persmissions/owner and call that before calling unpack in gneeral.

Reply:
hm... good point. Do you think it have sense to remove possibility to extend command line at all?